ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1484,16 +1495,54 @@ withConnectionManager ConnectionManagerArguments {
                          (mkTransition connState connState'))
 
               OutboundDupState connId connThread handle Ticking -> do
-                -- @
-                --   DemotedToCold^{Duplex}_{Local}
-                --     : OutboundState^\tau Duplex
-                --     → InboundIdleState^\tau Duplex
-                -- @
                 let connState' = InboundIdleState connId connThread handle Duplex
-                writeTVar connVar connState'
-                return (DemoteToColdLocalNoop
-                         (Just $ mkTransition connState connState')
-                         (abstractState $ Known connState'))
+                    tr = mkTransition connState connState'
+
+                numberOfConns <- countIncomingConnections state
```

Comment:
I think I understand, let me know if that's what you've been thinking about.  We count the actual number of inbound connections and guard each transition from an outbound state to duplex or inbound state.  I think that's the right approach.

Reply:
Exactly! I noticed that we were using something like `countPrunableConnections` instead which didn't make much sense since the hard limit is to limit the number of inbound connections. and `incomingConnections ⊆ prunableConnections` this is a good property to have because it tells us we don't have to worry much about needing a connection to prune.

Reply:
I am worried that we changed the language in some subtle way:

* inbound connections used be only connections that where accepted
* prunable ones where connections that were used by the server

it seems that you call what we used to call _prunable_ as _incoming_ now.

I prefer to use _prunable_ over _incoming_ because that's too easy to be confused with the connections that were accepted.

Reply:
Right, what I have been calling incoming connections is actually connections that either were accepted or in the inbound side.